src/garage/tf/algos/npo.py

```diff
@@ -338,18 +339,15 @@ def _build_inputs(self):
                 policy_state_info_vars[k] for k in self.policy.state_info_keys
             ]
 
-        # concat the action input with obs_var to become the final
-        # state input
         augmented_obs_var = obs_var
         for k in self.policy.state_info_keys:
             extra_state_var = policy_state_info_vars[k]
             extra_state_var = tf.cast(extra_state_var, tf.float32)
             augmented_obs_var = tf.concat([augmented_obs_var, extra_state_var],
                                           -1)
 
-        self.policy.build(augmented_obs_var)
-        self._old_policy.build(augmented_obs_var)
-        self._old_policy.model.parameters = self.policy.model.parameters
+        self.policy.build(augmented_obs_var, name='policy')
```

Comment:
wouldn't it be more canonical to capture the network here (as the return value) rather than always using `self.policy.networks['foo'])`? 

perhaps we should make `Model.networks` private

Reply:
Agree, we should make policies return the network when built, and cache it. Will fix that.